lib/puppet/type/file.rb

```diff
@@ -835,6 +835,9 @@ def stat
     rescue Errno::EACCES
       warning _("Could not stat; permission denied")
       nil
+    rescue Errno::EINVAL
+      warning _("Could not stat; invalid pathname")
```

Comment:
Is this warning useful/wanted?

Reply:
I think so. It's consistent with what's already there.